js/init.js

```diff
@@ -0,0 +1,3 @@
+if (typeof hideChoiceByEvent === 'undefined') {
+    var hideChoiceByEvent = {};
```

Comment:
I think `init.js` is not needed, since `setJsSetting()` function could be changed to `setJsSettings()` in order to handle `hideChoiceByEvent` initialization, e.g.:

```
protected function setJsSettings($value) {
    echo '<script>hideChoiceByEvent = ' . json_encode($value) . ';</script>';
}
```